workflow/rules/peak_analysis.smk

```diff
@@ -0,0 +1,209 @@
+    output:
+        # all output-files must share the same basename and only differ by it's extension
+        # Usable extensions (and which tools they implicitly call) are listed here:
+        #         https://snakemake-wrappers.readthedocs.io/en/stable/wrappers/macs2/callpeak.html.
+        multiext("results/macs2_callpeak/{sample}-{control}.narrow",
+                 "_peaks.xls",
+                 # these output extensions internally set the --bdg or -B option:
+                 "_treat_pileup.bdg",
+                 "_control_lambda.bdg",
+                 # these output extensions internally set the --broad option:
+                 "_peaks.narrowPeak",
+                 "_summits.bed"
+                 )
+    log:
+        "logs/macs2/callpeak.{sample}-{control}.narrow.log"
+    params: # ToDo: move to config?
+        "-f BAMPE -g hs --SPMR --qvalue 0.05 --keep-dup all"
+    wrapper:
+        "0.66.0/bio/macs2/callpeak"
+
+rule peaks_count:
+    input:
+        peaks="results/macs2_callpeak/{sample}-{control}.{peak}_peaks.{peak}Peak"
+    output:
+        "results/macs2_callpeak/peaks_count/{sample}-{control}.{peak}.peaks_count.tsv"
+    log:
+        "logs/macs2_callpeak/peaks_count/{sample}-{control}.{peak}.peaks_count.log"
+    conda:
+        "../envs/gawk.yaml"
+    shell:
+        "cat {input.peaks} | wc -l | gawk -v OFS='\t' '{{print \"{wildcards.sample}-{wildcards.control}_{wildcards.peak}_peaks\", $1}}' > {output} 2> {log}"
+
+rule sm_report_peaks_count_plot:
+    input:
+        get_peaks_count_plot_input()
+    output:
+        report("results/macs2_callpeak/plots/plot_{peak}_peaks_count.pdf", caption="../report/plot_peaks_count_macs2.rst", category="CallPeaks")
+    log:
+        "logs/macs2_callpeak/plot_{peak}_peaks_count.log"
+    conda:
+        "../envs/r_plots.yaml"
+    script:
+        "../scripts/plot_peaks_count_macs2.R"
+
+rule bedtools_intersect:
+    input:
+        left="results/orphan_rm_sorted/{sample}.bam",
+        right="results/macs2_callpeak/{sample}-{control}.{peak}_peaks.{peak}Peak"
+    output:
+        pipe("results/intersect/{sample}-{control}.{peak}.intersected.bed")
+    params:
+        extra="-bed -c -f 0.20"
+    log:
+        "logs/intersect/{sample}-{control}.{peak}.intersected.log"
+    wrapper:
+        "0.66.0/bio/bedtools/intersect"
+
+rule frip_score:
+    input:
+        intersect="results/intersect/{sample}-{control}.{peak}.intersected.bed",
+        flagstats="results/orphan_rm_sorted/{sample}.orphan_rm_sorted.flagstat"
+    output:
+        "results/intersect/{sample}-{control}.{peak}.peaks_frip.tsv"
+    log:
+        "logs/intersect/{sample}-{control}.{peak}.peaks_frip.log"
+    conda:
+        "../envs/gawk.yaml"
+    shell:
+        "grep 'mapped (' {input.flagstats} | gawk -v a=$(gawk -F '\t' '{{sum += $NF}} END {{print sum}}' < {input.intersect}) -v OFS='\t' "
+        "'{{print \"{wildcards.sample}-{wildcards.control}_{wildcards.peak}_peaks\", a/$1}}' > {output} 2> {log}"
```

Comment:
Same as above, I would make this rather long pipe with the extensive nested gawk stretch across more lines, e.g.:
```suggestion
        "grep 'mapped (' {input.flagstats} | "
        " gawk -v a=$(gawk -F '\t' '{{sum += $NF}} END {{print sum}}' < {input.intersect}) "
        "  -v OFS='\t' "
        " '{{print \"{wildcards.sample}-{wildcards.control}_{wildcards.peak}_peaks\", a/$1}}' "
        " > {output} 2> {log}"
```